docs/site/Behind-the-scene.md

```diff
@@ -1,22 +1,75 @@
 ---
 lang: en
-title: 'Components'
-keywords: LoopBack 4.0, LoopBack 4, Node.js, TypeScript, OpenAPI
+title: 'Behind the scene'
+keywords: LoopBack 4, Node.js, TypeScript, OpenAPI, Explanations, Concepts
 sidebar: lb4_sidebar
 permalink: /doc/en/lb4/Behind-the-scene.html
 ---
 
+LoopBack 4 defines some key building blocks to represent different
+responsibilities for typical API and/or Microservice applications.
+
+![Key concepts overview diagram](imgs/key-concepts-overview-diagram.png)
+
+- [**Application**](Application.md): The central class for setting up all of
+  your module’s components, controllers, servers and bindings. The Application
+  class extends [Context](Context.md) and provides the controls for starting and
+  stopping its associated servers.
+
+- [**Server**](Server.md): An implementation for inbound transports/protocols
+  such as REST (http, https), gRPC (http2) and graphQL (http, https). It
+  typically listens on a specific endpoint (protocol/host/port), handles
+  incoming requests, and then returns appropriate responses.
+
+- [**Controller**](Controllers.md): A class that implements operations defined
+  by the application’s REST API. It implements an application’s business logic
+  and acts as a bridge between the HTTP/REST API and domain/database models. A
+  Controller operates only on processed input and abstractions of backend
+  services / databases.
+
+- [**Interceptors**](Interceptors.md): A function that intercepts static or
+  instance method invocations on a class or object.
+
+- [**Route**](Routes.md): The mapping between your API specification and an
+  Operation. It tells LoopBack which Operation to `invoke()` when given an HTTP
+  request.
+
+- [**Sequence**](Sequence.md): A stateless grouping of
+  [Actions](Sequence.md#actions) that control how a Server responds to requests.
```

Comment:
Maybe we should use a consistent order from top to bottom, from left to right? For example:

Application/Component/Context/Binding
Server/Sequence/Action/Route/Controller/Repository/Service/Model/Relation/DataSource

Reply:
I see your point.

Personally, I would prefer sidebar sections to be sorted lexicographically (from A to Z) to make them easy to find. I guess we can use different sorting inside the markdown page.

I'd also like to discuss how to group the concepts along functional areas (Building REST APIs, Working with Data, etc.), I think this grouping may affect the way how we want to present the concepts on this page too.

I'll open a follow-up issue to discuss & improve this in a holistic way.

Reply:
See https://github.com/strongloop/loopback-next/issues/5777 and https://github.com/strongloop/loopback-next/issues/5776